src/CriteriaCollection.php

```diff
@@ -2,23 +2,26 @@
 
 namespace EthicalJobs\Storage;
 
-/**
- * Collection of criteria
- *
- * @author Andrew McLagan <andrew@ethicaljobs.com.au>
- */
 class CriteriaCollection extends Collection
 {
     /**
      * Push a criteria class
      *
-     * @param string $criteria
+     * Unclear what the purpose of this is, why it doesn't push but instead puts or why it was made a dependency.
+     *
+     * @TODO Remove dependency on this and potentially entire package
+     *
+     * @param  mixed  $values [optional]
      * @return $this
      */
-    public function push($criteria)
+    public function push(...$values)
     {
-        $instance = new $criteria;
+        foreach ($values as $value) {
+            $instance = new $value;
+
+            $this->put($value, $instance);
+        }
```

Comment:
This appears to be a breaking change, is it necessary to change?

Reply:
as far as I can tell, it's actually non-breaking. e.g.

```php
existing usage:
$firstVal = CoolClass::class
$secondVal = SickClass::class
$result1 = $criteriaCollection->push($singleValue)->push($singleValue)
$result1 : (CriteriaCollection[CoolClass::class => (obj)CoolClass, SickClass::class => (obj)SickClass)

this new usage:
$firstVal = CoolClass::class
$secondVal = SickClass::class
$result2 = $criteriaCollection->push($firstVal, $secondVal)
$result2 : (CriteriaCollection[CoolClass::class => (obj)CoolClass, SickClass::class => (obj)SickClass)

$result1 === $result2
```

does my assumption seem correct? that the content of the code is not a breaking change - passing one item alone is fine as is passing many?

Reply:
No you're right, it's non-breaking